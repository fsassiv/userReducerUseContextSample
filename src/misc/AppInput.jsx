import React from "react";
import { FormControl, Input } from "@material-ui/core";
import "./AppInput.scss";

const AppInput = ({
  inputPlaceholder,
  handleChange,
  inputTarget,
  inputValue
}) => {
  return (
    <FormControl>
      <Input
        onChange={({ target }) =>
          handleChange({ target: inputTarget, value: target.value })
        }
        placeholder={inputPlaceholder}
        value={inputValue}
        type={inputTarget === "confirmPassword" ? "password" : inputTarget}
        className="appInput"
      />
    </FormControl>
  );
};

export default AppInput;
