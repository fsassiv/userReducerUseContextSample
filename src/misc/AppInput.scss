.appInput {
  margin-bottom: 4%;
  background-color: white;
}
