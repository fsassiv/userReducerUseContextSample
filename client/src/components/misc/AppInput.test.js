import AppInput from "./AppInput.jsx";

describe("AppInput", () => {
  const mockChange = jest.fn();
  const defaultProps = {
    inputPlaceholder: "Digite aqui",
    handleChange: mockChange,
    inputValue: ""
  };
  describe("AppInput [text]", () => {
    const inputType = { inputTarget: "text" };

    const appInput = shallow(<AppInput {...defaultProps} {...inputType} />);

    it("rendered correctly", () => {
      expect(appInput).toMatchSnapshot();
    });

    describe("when typing into the AppInput...", () => {
      beforeEach(() => {
        appInput
          .find(".appInput")
          .simulate("change", { target: { value: "Texto teste" } });
      });

      it("call 'handleChange' callback", () => {
        expect(mockChange).toHaveBeenCalled();
      });
    });
  });

  describe("AppInput [password]", () => {
    const inputType = { inputTarget: "password" };
    const appInput = shallow(<AppInput {...defaultProps} {...inputType} />);

    it("rendered correctly", () => {
      expect(appInput).toMatchSnapshot();
    });

    describe("when typing into the AppInput...", () => {
      beforeEach(() => {
        appInput
          .find(".appInput")
          .simulate("change", { target: { value: "senha" } });
      });

      it("call 'handleChange' callback", () => {
        expect(mockChange).toHaveBeenCalled();
      });
    });
  });
});
