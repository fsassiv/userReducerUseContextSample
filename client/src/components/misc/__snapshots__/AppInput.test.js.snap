// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInput AppInput [password] rendered correctly 1`] = `ShallowWrapper {}`;

exports[`AppInput AppInput [text] rendered correctly 1`] = `ShallowWrapper {}`;
